File: pdf/templates/test/fixtures/_test.number.R
options( digits = 16 );
library( jsonlite );


<%= parameters.map( function( p ) { return p.name + ' = ' + p.default } ).join( '\n' ) %>
x = c( -5, -2.5, 0, 2.5, 5 )
y = <%= rName %>( x, <%= parameters.map( function( p ) { return p.name } ).join( ',' ) %> )

cat( y, sep = ",\n" )

data = list(
	<%= parameters.map( function( p ) { return p.name + ' = ' + p.name } ).join( ',\n' ) %>,
	data = x,
	expected = y
)

write( toJSON( data, digits = 16, auto_unbox = TRUE ), "./test/fixtures/number.json" )

File: mgf/templates/test/fixtures/_test.number.jl
using Distributions
using JSON

<%= parameters.map( function( p ) { return p.name + ' = ' + p.default } ).join( '\n' ) %>
d = <%= rName %>( <%= parameters.map( function( p ) { return p.name } ).join( ',' ) %> )

x = [ -5, -2.5, 0, 2.5, 5 ]

dmgf(t) = mgf(d, t )
y = map( dmgf, x )
println( y )

data = Dict([
	<%- parameters.map( function( p ) { return '("' + p.name + '", ' + p.name + ")" } ).join( ',\n\t' ) %>,
	("data", x),
	("expected", y)
])

outfile = open("./test/fixtures/number.json", "w")
JSON.json(data)

write( outfile, JSON.json(data) )
